### ISSUE TITLE
Minetest crashes when starting, somehow invalid lines get added to auth.txt

### ISSUE THREAD

**Issue by gpcf** [minetest/minetest#4602]
Sometimes, my minetest server crashes when starting, with the following traceback:

```
2016-10-08 11:42:46: ERROR[Main]: ModError: Failed to load and run script from /usr/share/games/minetest/builtin/init.lua:
2016-10-08 11:42:46: ERROR[Main]: /usr/share/games/minetest/builtin/game/auth.lua:48: Invalid line in auth.txt: "luiz:#1#AAAAAAAAAAAAAAAAAA"
2016-10-08 11:42:46: ERROR[Main]: stack traceback:
2016-10-08 11:42:46: ERROR[Main]:       [C]: in function 'error'
2016-10-08 11:42:46: ERROR[Main]:       /usr/share/games/minetest/builtin/game/auth.lua:48: in function 'read_auth_file'
2016-10-08 11:42:46: ERROR[Main]:       /usr/share/games/minetest/builtin/game/auth.lua:82: in main chunk
2016-10-08 11:42:46: ERROR[Main]:       [C]: in function 'dofile'
2016-10-08 11:42:46: ERROR[Main]:       /usr/share/games/minetest/builtin/game/init.lua:20: in main chunk
2016-10-08 11:42:46: ERROR[Main]:       [C]: in function 'dofile'
2016-10-08 11:42:46: ERROR[Main]:       /usr/share/games/minetest/builtin/init.lua:38: in main chunk

```

This error persists until the offending line is removed from auth.txt
I'm running the Minetest version that comes with Debian unstable, 0.4.14

**Comment by est31** [comment]
@gpcf is the invalid line the last one in the file? It might have came from an invalid exit of the server (like a crash, or sending it sigkill instead of sigint like it'd be done properly).

**Comment by gpcf** [comment]
@est31 Sorry, I fixed the file using a sed script, but it happens quite often, I'll tell you the next time it happens.

**Comment by gpcf** [comment]
@est31 yes, It's always the last line, i just found old auth files. 

These are the last two lines of the auth.txt file that caused the crash:

```
Hogan4455:#1#AQAAAD4AAAAAAAAAQAAAAA#MNonROBIyrZ9TJnx8evBG9On9NKMMMPdfi7xwrDj6sMvdefPR0pE6r8P75M+PL08ZpC5ZD0tsvx23+26LplspXJEFHnkUTznWjXWTDoJpzdSNVGYMBo0FedHYq9rbXtxTbWT0bweYmskpQkg3R/8MiJNjCkFZnXV2+HbiqY2gRlELnO5zljf5v+Q67hjkbM8ZioWG5+4Qegp+7+U8DlHJj+DlNs3E8qJblJqUCJ/4bshk7xrMOeiYZeAIrqXz+iLUPt5KXxKF990nkUpzI7pVoG0WL5oN5l33lyg8m1fqYbQIpXUQoLlnGt2bsh9KKftVFeKu0NvNjOL7RksAt8EYg:interact,shout:1473390010
luiz:#1#AAAAAAAAAAAAAAAAAA
´´´

**Comment by paramat** [comment]
Yes the problem seems to be a corrupted auth.txt file.

**Comment by gpcf** [comment]
Yeah. The problem is also that this is a fatal error even though it just could ignore the line, logging the incident, as I have proposed in the pull request #4603. What do you think?

**Comment by est31** [comment]
Yeah it seems its the manifestation of the general bug "text file easily corrupts" that. It'd be fixed to store it all in a sqlite or some other database, but all attempts to implement such things failed in the past due to the opposition by some devs (hmmm, etc) who prefer to use files due to simplicity...

**Comment by gpcf** [comment]
Well, LevelDB also corrupts now and then, we had some problems on our server, then we migrated it to SQLite, never had a problem since. What do you think of my idea of making the error non-fatal?

**Comment by est31** [comment]
@gpcf see my comment in #4603. The way to respond to these corruptions is to play back backups, which usually is a manual step, and having a fatal error is required to cause manual intervention.

**Comment by gpcf** [comment]
Ok, I get your point.

**Comment by gpcf** [comment]
I'm working on a real fix that checks the file written for consistency, if it's not consistent, it rewrites it.

**Comment by nerzhul** [comment]
We will fix that in the next release when auth will be ported to a specific database

**Comment by SmallJoker** [comment]
This issue is fixed since fc9747e. (`core.safe_file_write`)
